### PR TITLE
Merge release/1.2.0 into develop

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

As the title says, merge release/1.2.0 into jcsda_emc_spack_stack. All changes have been reviewed, tested and approved in the release branches. CI tests will be run again, see below for more information.

Associated spack PR needs to be merged first: https://github.com/NOAA-EMC/spack/pull/208

## Testing

Only rerunning CI tests, manual testing on Dom's macOS has been done for both develop and release/1.2.0 independently beforehand, the release branch was tested on all supported platforms, and develop was tested at least on Hera for the new ufs-weather-model-static template (static libraries).

CI tests passed for [35490e4](https://github.com/NOAA-EMC/spack-stack/pull/421/commits/35490e468145da89beae5b1ee873c335a67d4e41). Follow-up commits are cleanup only, no code changes.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,8 +2,8 @@
   path = spack
   #url = https://github.com/spack/spack
   #branch = develop
-  url = https://github.com/NOAA-EMC/spack
-  branch = jcsda_emc_spack_stack
+  url = https://github.com/climbfuji/spack
+  branch = feature/merge_release_120_into_jcsda_emc_spack_stack
```
Comment (climbfuji) [inline .gitmodules]: Note to self. Revert after https://github.com/NOAA-EMC/spack/pull/208 was merged

File: configs/containers/docker-ubuntu-clang-mpich.yaml [hunk 2]
```diff
@@ -95,7 +96,7 @@ spack:
       os: ubuntu:20.04
       spack:
         url: https://github.com/noaa-emc/spack
-        ref: jcsda_emc_spack_stack
+        ref: spack-stack-1.2.0
```
Comment (climbfuji) [inline configs/containers/docker-ubuntu-clang-mpich.yaml]: Note to self. Revert after CI tests completed (this is not used in CI)

File: configs/containers/docker-ubuntu-gcc-openmpi.yaml [hunk 3]
```diff
@@ -82,7 +83,7 @@ spack:
       os: ubuntu:20.04
       spack:
         url: https://github.com/noaa-emc/spack
-        ref: jcsda_emc_spack_stack
+        ref: spack-stack-1.2.0
```
Comment (climbfuji) [inline configs/containers/docker-ubuntu-gcc-openmpi.yaml]: Note to self. Revert after CI tests completed (this is not used in CI)

File: configs/containers/docker-ubuntu-intel-impi.yaml [hunk 4]
```diff
@@ -96,7 +101,7 @@ spack:
       os: ubuntu:20.04
       spack:
         url: https://github.com/noaa-emc/spack
-        ref: jcsda_emc_spack_stack
+        ref: spack-stack-1.2.0
```
Comment (climbfuji) [inline configs/containers/docker-ubuntu-intel-impi.yaml]: Note to self. Revert after CI tests completed (this is not used in CI)

File: configs/templates/skylab-3.0.0/spack.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
+spack:
```
Comment (climbfuji) [inline configs/templates/skylab-3.0.0/spack.yaml]: Note to self. Delete this file after CI tests completed (this is not used in CI, and belongs only to the release branch)

**Review by shlyaeva — APPROVED** [review]

**Review by srherbener — APPROVED** [review]

**Pull request merged**